app/models/constituency.rb

```diff
@@ -46,4 +50,69 @@ def sitting_mp?
   def mp_url
     "#{MP_URL}/#{mp_name.parameterize}/#{mp_id}"
   end
+
+  def example_postcode
+    super || fetch_and_save_example_postcode
+  end
+
+  def reset_example_postcode
+    update(example_postcode: nil)
+  end
+
+  private
+
+  def fetch_and_save_example_postcode
+    area = fetch_area(ons_code)
+
+    if area
+      postcode = fetch_example_postcode(area["id"])
+    else
+      postcode = nil
+    end
+
+    if postcode
+      update(example_postcode: postcode)
+    end
+
+    postcode
+
+  rescue Faraday::Error => e
+    Appsignal.send_exception(e) if defined?(Appsignal)
+    return nil
+  end
+
+  def faraday
+    @faraday ||= Faraday.new(MAPIT_HOST) do |f|
+      f.response :follow_redirects
+      f.response :raise_error
+      f.adapter  :net_http_persistent
+    end
+  end
+
+  def get(path)
+    faraday.get(path) do |request|
+      request.options[:timeout] = 5
+      request.options[:open_timeout] = 5
+    end
+  end
+
+  def fetch_area(ons_code)
+    response = get(MAPIT_AREA_URL % { ons_code: ons_code })
+
+    if response.success?
+      JSON.load(response.body)
+    else
+      nil
+    end
+  end
+
+  def fetch_example_postcode(area_id)
+    response = get(MAPIT_POSTCODE_URL % { area_id: area_id })
+
+    if response.success?
+      JSON.load(response.body).upcase.gsub(/\s+/, "")
```

Comment:
Minor, but we could push this through the `PostcodeSanitizer` so it's more obvious what is happening?